File: framework/ui/src/io/intino/alexandria/ui/displays/components/Authenticate.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.events.SignEvent;
import io.intino.alexandria.ui.displays.events.SignListener;
import io.intino.alexandria.ui.displays.notifiers.AuthenticateNotifier;

public class Authenticate<DN extends AuthenticateNotifier, B extends Box> extends AbstractAuthenticate<DN, B> {
	private SignListener executeListener;

	public Authenticate(B box) {
		super(box);
	}

	public void onExecute(SignListener listener) {
		this.executeListener = listener;
	}

	public void info(String value) {
		executeListener.accept(new SignEvent(this, null, null, null));
	}

	public void execute() {
		// TODO
	}

}

File: framework/ui-elements/src/displays/components/Authenticate.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractAuthenticate from "../../../gen/displays/components/AbstractAuthenticate";
import AuthenticateNotifier from "../../../gen/displays/notifiers/AuthenticateNotifier";
import AuthenticateRequester from "../../../gen/displays/requesters/AuthenticateRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';
import { withSnackbar } from 'notistack';

const styles = theme => ({});

class Authenticate extends AbstractAuthenticate {

	constructor(props) {
		super(props);
		this.notifier = new AuthenticateNotifier(this);
		this.requester = new AuthenticateRequester(this);
	};

    sign = () => {
        window.crypto.subtle.signText("Hola")
    };

}

export default withStyles(styles, { withTheme: true })(withSnackbar(Authenticate));
DisplayFactory.register("Authenticate", withStyles(styles, { withTheme: true })(withSnackbar(Authenticate)));